src/main/scala/nl/knaw/dans/easy/fedoratobag/filter/package.scala

```diff
@@ -36,32 +36,23 @@ package object filter {
       fileInfos.exists(_.isOriginal) && fileInfos.exists(!_.isOriginal)
     }
 
-    def selectForFirstBag(emd: Node, hasSecondBag: Boolean, europeana: Boolean, noPayload: Boolean = false): Try[List[FileInfo]] = {
+    def selectForFirstBag(emd: Node, hasSecondBag: Boolean, europeana: Boolean, noPayload: Boolean = false): Try[List[FileInfo]] = Try {
```

Comment:
Drop the tries from this line, nothing can cause a failure. Thus you can drop the for-comprehension in `getInfoFirstBag` too.